vocode/streaming/streaming_conversation.py

```diff
@@ -1375,15 +1400,27 @@ def get_duration_spoken_seconds(time_started_speaking: float | None):
             return 0
         return time.perf_counter() - time_started_speaking
 
+    @staticmethod
+    def set_message_sent_on_interrupt(
+        duration_spoken_seconds: float,
+        json_transcript_entry: StateAgentTranscriptMessage,
+        synthesis_result: SynthesisResult,
+    ):
+        json_transcript_entry.message_sent = synthesis_result.get_message_up_to(
+            duration_spoken_seconds
+        )
```

Comment:
need to add the null case handling. Shouldn't ever be None but its typed that way in send speech to output. Going to double check this, but can't say its not None because `produce_interruptible_agent_response_event_nonblocking` is used for more things than just the agent responses.